src/Sparrow/ProcessorInfo.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using Sparrow.Logging;
+using Sparrow.Platform;
+using Sparrow.Platform.Posix;
+using Voron.Platform.Posix;
+
+namespace Sparrow
+{
+    public static class ProcessorInfo
+    {
+        private static Logger _logger = LoggingSource.Instance.GetLogger("ProcessorInfo", "Raven/Server");
+
+        public static int ProcessorCount
+        {
+            get
+            {
+                if (PlatformDetails.RunningOnPosix == false)
+                    return Environment.ProcessorCount;
+
```

Comment:
Should probably cache that, I don't think we need to support variable number of cores.